app/api/routes/user.py

```diff
@@ -1,209 +1,111 @@
-from datetime import datetime, timezone
-from uuid import UUID, uuid4
 from fastapi import APIRouter, Depends, HTTPException, status
-from app.api.routes.dependencies import get_current_active_user, get_current_admin, get_current_user
-from models import ChangeRoleRequest, GetUserResponseModel, UserCreateRequestModel, User, UserCreateResponseModel, UserUpdateRequestModel, UserUpdateResponseModel 
-from api.auth_utlis import get_password_hash, is_valid_password
-
+from sqlalchemy import UUID
+from sqlalchemy.orm import Session
+from app.api.routes import dependencies
+from app.services import user_service, order_service
+from ... import models,schemas, database
 
 router = APIRouter()
 
+@router.post("/users/", response_model=schemas.UserCreateResponseModel, status_code=status.HTTP_201_CREATED)
+def create_user(user: schemas.UserCreateRequestModel, db: Session = Depends(database.get_db)):
+   return user_service.create_user(db=db,user=user)  
+
+
+@router.get("/users/{user_id}", response_model=schemas.GetUserResponseModel, status_code=status.HTTP_200_OK)
+async def get_user_details(
+    user_id: UUID, 
+    db: Session = Depends(database.get_db), 
+    current_user: models.User = Depends(dependencies.get_current_user)
+):
+    user = user_service.get_user_by_id(user_id, db)
```

Comment:
We should move this after line 27 
If the current user is not admin and the current user id != user_id then we don't need to call the database at all 

I just noticed that

Reply:
Yes thank you i appreciate this, only query the database if access is allowed.
Logic optimized.